synapse/lib/coro.py

```diff
@@ -241,21 +241,35 @@ def execspawn():
         proc.terminate()
         raise
 
-# shared process pool
-forkpool = None
+forkpool = None     # general use shared process pool
+_parserpool = None  # storm parser dedicated process pool
 if multiprocessing.current_process().name == 'MainProcess':
-    # only create the forkpool in the MainProcess...
+    # only create the forkpools in the MainProcess...
     try:
         mpctx = multiprocessing.get_context('forkserver')
-        max_workers = int(os.getenv('SYN_FORKED_WORKERS', 1))
+        _parserpool = concurrent.futures.ProcessPoolExecutor(mp_context=mpctx, max_workers=1)
+        atexit.register(_parserpool.shutdown)
+    except OSError as e:  # pragma: no cover
+        logger.warning(f'Failed to init Storm parser forkserver pool, fallback enabled: {e}', exc_info=True)
+
+    try:
+        mpctx = multiprocessing.get_context('forkserver')
+        max_workers = int(os.getenv('SYN_FORKED_WORKERS', 0)) or None
```

Comment:
This env var was/is undocumented, but it is a change in behavior. If this will be released while we are still supported py < 3.9 we may want to set a low default instead of None since _a lot_ of idle processes could be spun up.